keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -88,6 +90,51 @@ class SqlDeliveryConfigRepository(
     return deliveryConfigUIDs.size
   }
 
+  override fun delete(name: String) {
+    val deliveryConfigUid: String = jooq
+      .select(DELIVERY_CONFIG.UID)
+      .from(DELIVERY_CONFIG)
+      .where(DELIVERY_CONFIG.NAME.eq(name))
+      .fetchOne(DELIVERY_CONFIG.UID)
+
+    jooq
+      .deleteFrom(DELIVERY_CONFIG)
+      .where(DELIVERY_CONFIG.NAME.eq(name))
+      .execute()
+    jooq
+      .deleteFrom(DELIVERY_CONFIG_LAST_CHECKED)
+      .where(DELIVERY_CONFIG_LAST_CHECKED.DELIVERY_CONFIG_UID.eq(deliveryConfigUid))
+      .execute()
+  }
+
+  override fun deleteResourceFromEnv(deliveryConfigName: String, environmentName: String, resourceId: ResourceId) {
+    jooq.deleteFrom(ENVIRONMENT_RESOURCE)
+      .where(ENVIRONMENT_RESOURCE.ENVIRONMENT_UID.eq(envUid(deliveryConfigName, environmentName)))
+      .and(ENVIRONMENT_RESOURCE.RESOURCE_UID.eq(resourceId.uid))
+      .execute()
+  }
+
+  override fun deleteEnvironment(deliveryConfigName: String, environmentName: String) {
+    val deliveryConfigUid = getDeliveryConfigId(deliveryConfigName)
+    jooq
+      .deleteFrom(ENVIRONMENT_ARTIFACT_CONSTRAINT)
+      .where(ENVIRONMENT_ARTIFACT_CONSTRAINT.ENVIRONMENT_UID.eq(envUid(deliveryConfigName, environmentName)))
```

Comment:
Could move `envUid(deliveryConfigName, environmentName)` above the jooq methods so you don't call it repeatedly.

Reply:
Agree, since envUid() is a db subquery, it's not free to repeat.